source/MRViewer/MRPlaneWidget.cpp

```diff
@@ -0,0 +1,118 @@
+#include "MRPlaneWidget.h"
+#include "MRMesh/MRObjectsAccess.h"
+#include "MRMesh/MRObjectMeshHolder.h"
+#include "MRMesh/MRMakePlane.h"
+#include "MRMesh/MRMesh.h"
+#include "MRMesh/MRObjectMesh.h"
+#include "MRMesh/MRObjectLines.h"
+
+namespace MR
+{
+
+PlaneWidget::PlaneWidget( const Plane3f& plane, const Box3f& box, std::function<void( const Plane3f& )> onPlaneUpdate )
+: plane_(plane),
+box_(box),
+onPlaneUpdate_( onPlaneUpdate )
+{ 
+    std::shared_ptr<Mesh> planeMesh = std::make_shared<Mesh>( makePlane() );
+    planeObj_ = std::make_shared<ObjectMesh>();
+    planeObj_->setName( "PlaneObject" );
+    planeObj_->setMesh( planeMesh );
+    planeObj_->setAncillary( true );
+    planeObj_->setFrontColor( Color( Vector4f::diagonal( 0.3f ) ), false );
+    planeObj_->setBackColor( Color( Vector4f::diagonal( 0.3f ) ) );
+    SceneRoot::get().addChild( planeObj_ );
```

Comment:
better dont setup plane in constructor to have `defined/undefined` logic based on `bool(planeObj_)`